### PR TITLE
[MMI] Handle personal sign and sign typed operations

### PR DESCRIPTION
**Pull request by zone-live**
This PR adds logic for MMI, after [updating](https://github.com/MetaMask/core/pull/1506) the signature controller in Core repo, we need to update the extension side. Where we are adding our `deferSetAsSigned` param before calling for the signing methods, and also listening for signing updates using the signature-controller hub.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/modals/modal.js [hunk 1]
```diff
@@ -439,7 +439,7 @@ function mapDispatchToProps(dispatch) {
     },
     ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
     setWaitForConfirmDeepLinkDialog: (wait) =>
-      dispatch(mmiActions.setWaitForConfirmDeepLinkDialog(wait)),
+      mmiActions.setWaitForConfirmDeepLinkDialog(wait),
```
Comment (albertolive) [inline ui/components/app/modals/modal.js:442]: Are you sure we don't need the dispatch, here?

File: ui/components/app/signature-request-original/signature-request-original.container.js [hunk 2]
```diff
@@ -145,8 +140,7 @@ function mmiMapDispatchToProps(dispatch) {
         }),
       ),
     setWaitForConfirmDeepLinkDialog: (wait) =>
-      dispatch(mmiActions.setWaitForConfirmDeepLinkDialog(wait)),
```
Comment (albertolive) [inline ui/components/app/signature-request-original/signature-request-original.container.js:148]: Same, are you sure we don't need the dispatch?

File: ui/components/app/signature-request-original/signature-request-original.container.js [hunk 3]
```diff
@@ -202,25 +195,17 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
...
-            unapprovedPersonalMessages,
-          );
-          id = msgData.custodyId;
-        }
+        await dispatchProps.resolvePendingApproval(_msgData.id);
+        dispatchProps.completedTx(_msgData.id);
+
         dispatchProps.showCustodianDeepLink({
-          custodyId: id,
+          custodyId: null,
```
Comment (albertolive) [inline ui/components/app/signature-request-original/signature-request-original.container.js:202]: Why do we need to pass null here?

File: ui/components/app/signature-request-original/signature-request-original.container.js (hunk 2 above)
Comment (zone-live) [reply ui/components/app/signature-request-original/signature-request-original.container.js:148]: Answered above 👍🏼

File: ui/components/app/signature-request-original/signature-request-original.container.js (hunk 3 above)
Comment (zone-live) [reply ui/components/app/signature-request-original/signature-request-original.container.js:202]: We don't have the custodyId at this point, so it goes as null.

**Comment by zone-live** [comment]
@metamaskbot update-policies

File: ui/components/institutional/custody-confirm-link-modal/custody-confirm-link-modal.js [hunk 4]
```diff
@@ -72,8 +72,8 @@ const CustodyConfirmLink = ({ hideModal }) => {
       category: MetaMetricsEventCategory.MMI,
       event: MetaMetricsEventName.DeeplinkClicked,
     });
-    dispatch(mmiActions.setWaitForConfirmDeepLinkDialog(false));
-    dispatch(hideModal());
+    mmiActions.setWaitForConfirmDeepLinkDialog(false);
+    hideModal();
```
Comment (NidhiKJha) [inline ui/components/institutional/custody-confirm-link-modal/custody-confirm-link-modal.js:76]: is dispatch not required for `hideModal()` ?

Comment (zone-live) [reply ui/components/institutional/custody-confirm-link-modal/custody-confirm-link-modal.js:76]: Doesn't need it, as its being referenced by the parent here: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/modals/modal.js#L401

**Review by albertolive — DISMISSED** [review]

File: ui/store/actions.ts [hunk 5]
```diff
@@ -3511,10 +3511,13 @@ export function resolvePendingApproval(
     await submitRequestToBackground('resolvePendingApproval', [id, value]);
     // Before closing the current window, check if any additional confirmations
     // are added as a result of this confirmation being accepted
+
+    ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
```
Comment (cryptotavares) [inline ui/store/actions.ts:3515]: won't this cause the lint to fail on mmi build? (lint after codefence)

File: package.json [hunk 6]
```diff
@@ -261,7 +261,7 @@
     "@metamask/rpc-methods-flask": "npm:@metamask/rpc-methods@0.37.1-flask.1",
     "@metamask/safe-event-emitter": "^2.0.0",
     "@metamask/scure-bip39": "^2.0.3",
-    "@metamask/signature-controller": "^4.0.1",
```
Comment (cryptotavares) [inline package.json:264]: There was patch on `@metamask/signature-controller`. I believe that it is still needed, right @matthewwalsh0?
Either way, if it is still required, you can remove and re-add a patch to the v5.3.0. If it is not required anymore, you can remove the patch

File: ui/store/actions.ts (hunk 5 above)
Comment (zone-live) [reply ui/store/actions.ts:3515]: Those will be taken cared in a next PR 👍🏼

**Review by cryptotavares — APPROVED** [review]

**Review by shane-t — APPROVED** [review]

**Pull request merged**